GitHub Actions CI failing due to Composer constraint mismatch (drupal/forum vs drupal/core)

### Description
Our GitHub Actions pipeline is currently failing during the Composer package resolution step. The failure is caused by a dependency conflict between `drupal/forum`, `drupal/history`, and the version of `drupal/core` provided by the local path repository (`10.5.x-dev`).

### Error Logs
The CI runner outputs the following Composer resolution error:

```text
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires drupal/forum ^1.0 -> satisfiable by drupal/forum[1.0.0, ..., 1.x-dev].
    - drupal/forum 1.0.0 requires drupal/core ^8 -> satisfiable by drupal/core[8.0.0-beta6, ..., 8.9.x-dev] from composer repo ([https://repo.packagist.org](https://repo.packagist.org)) but drupal/core[10.5.x-dev] from path repo (core) has higher repository priority. The packages from the higher priority repository do not match your constraint and are therefore not installable. That repository is canonical so the lower priority repo's packages are not installable. See [https://getcomposer.org/repoprio](https://getcomposer.org/repoprio) for details and assistance.
    - drupal/forum[1.0.1, ..., 1.x-dev] require drupal/history * -> satisfiable by drupal/history[dev-1.x, 1.0.0, 1.x-dev].
    - drupal/history[dev-1.x, 1.0.0, ..., 1.x-dev] require drupal/core >11.3 -> satisfiable by drupal/core[dev-main, 11.3.1, ..., 11.x-dev, 12.x-dev (alias of dev-main)] from composer repo ([https://repo.packagist.org](https://repo.packagist.org)) but drupal/core[10.5.x-dev] from path repo (core) has higher repository priority. The packages from the higher priority repository do not match your constraint and are therefore not installable. That repository is canonical so the lower priority repo's packages are not installable. See [https://getcomposer.org/repoprio](https://getcomposer.org/repoprio) for details and assistance.

Error: Process completed with exit code 2.

--- a/.github/workflows/php.yml
+++ b/.github/workflows/php.yml
@@ -78,6 +78,19 @@ jobs:
         composer require wikimedia/composer-merge-plugin
         composer config --json extra.merge-plugin.require '["profiles/contrib/apigee_devportal_kickstart/composer.json"]'
         composer config platform.php ${{ matrix.php-version }}
+        
+        # --- CI FIX FOR DRUPAL 10 AND 11 MONOLITHIC REPOS ---
+        # Modify composer.json directly using jq to bypass composer config CLI limits
+        
+        # Drupal 10 still has Forum and Tour in core.
+        if [[ "${{ matrix.drupal-core }}" == 10.* ]]; then
+          jq '.replace += {"drupal/forum": "*", "drupal/tour": "*"}' composer.json > tmp.json && mv tmp.json composer.json
+        fi
+        
+        # Drupal 10 and 11 both still have History in core.
+        # We must prevent the contrib Forum module from trying to download the contrib History module.
+        jq '.replace += {"drupal/history": "*"}' composer.json > tmp.json && mv tmp.json composer.json
+
         composer update --with-all-dependencies
         composer require --dev phpspec/prophecy-phpunit:^2
 

--- a/.github/workflows/php.yml
+++ b/.github/workflows/php.yml
@@ -79,15 +79,11 @@ jobs:
         composer config --json extra.merge-plugin.require '["profiles/contrib/apigee_devportal_kickstart/composer.json"]'
         composer config platform.php ${{ matrix.php-version }}
         
-        # --- CI FIX FOR DRUPAL 10 AND 11 MONOLITHIC REPOS ---
-        # Modify composer.json directly using jq to bypass composer config CLI limits
-        
         # Drupal 10 still has Forum and Tour in core.
         if [[ "${{ matrix.drupal-core }}" == 10.* ]]; then
           jq '.replace += {"drupal/forum": "*", "drupal/tour": "*"}' composer.json > tmp.json && mv tmp.json composer.json
         fi
         
-        # Drupal 10 and 11 both still have History in core.
         # We must prevent the contrib Forum module from trying to download the contrib History module.
         jq '.replace += {"drupal/history": "*"}' composer.json > tmp.json && mv tmp.json composer.json
 

